### PR TITLE
GROK-13664: EDA: Dimension reduction

### PR DESCRIPTION
**Pull request by vmakarichev**
UMAP, t-SNE & SPE and their call via Top Menu are added. Code refactoring is provided.

### CODE REVIEW

File: packages/EDA/src/eda-tools.ts [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+// Principal components analysis (PCA)
+export async function computePCA(table: DG.DataFrame, features: DG.ColumnList, components: number,
+  center: boolean, scale: boolean): Promise<DG.DataFrame> 
+{
+  checkWasmDimensionReducerInputs(features, components);
+
+  const centerNum = center ? 1 : 0;
+  const scaleNum = scale ? 1 : 0;
+
+  let _output: any;
```
Comment (skalkin) [inline packages/EDA/src/eda-tools.ts:25]: It's not "any", right?

File: packages/EDA/src/eda-tools.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+      worker.terminate();
+      resolve(e.data.embeddings);    
+  }}); 
+  
+  await promise.then(
+    result => { workerOutput = result; },
+    error => { throw new Error ('applying UMAP fails.'); }
+  );
+
+  const embeddings = workerOutput as any[][];
```
Comment (skalkin) [inline packages/EDA/src/eda-tools.ts:89]: Why "any"?

**Review by skalkin — APPROVED** [review]

File: packages/EDA/src/eda-tools.ts (hunk 1 above)
Comment (vmakarichev) [reply packages/EDA/src/eda-tools.ts:25]: Sure!

File: packages/EDA/src/eda-tools.ts (hunk 2 above)
Comment (vmakarichev) [reply packages/EDA/src/eda-tools.ts:89]: Fixed!

**Pull request merged**